sample/src/App.fs

```diff
@@ -73,6 +73,30 @@ module ReactLib =
                 ]
             ]
         ]
+module LitComponents = 
+    open Lit.Extensions
+    open Fable.Core
+
+    [<AttachMembers>]
+    type CounterSample() =
+        inherit LitElement()
+
+        let mutable counter = 0
+        let mutable nameAttribute = "World"
+
+        static member properties =
+            {| counter = {| state = true |}
```

Comment:
This is a complex one for me to better decide how to implement with type safety

those optional members are really annoying to deal with and anonymous objects just feel nicer IMO but I know not everyine likes this

I tried to define it like this 
```fsharp

///Defines options for a property accessor.
type PropertyDeclaration =


    /// When set to `true`, indicates the property is internal private state. The
    /// property should not be set by users. When using TypeScript, this property
    /// should be marked as `private` or `protected`, and it is also a common
    /// practice to use a leading `_` in the name. The property is not added to
    /// `observedAttributes`.
    abstract member state : bool option

    ///
    /// Indicates how and whether the property becomes an observed attribute.
    /// If the value is `false`, the property is not added to `observedAttributes`.
    /// If true or absent, the lowercased property name is observed (e.g. `fooBar`
    /// becomes `foobar`). If a string, the string value is observed (e.g
    /// `attribute: 'foo-bar'`).
    ///
    abstract member attribute : string option

    /// Indicates the type of the property. This is used only as a hint for the
    /// `converter` to determine how to convert the attribute
    /// to/from a property.
    abstract member ``type`` : obj option

    /// Indicates how to convert the attribute to/from a property. If this value
    /// is a function, it is used to convert the attribute value a the property
    /// value. If it's an object, it can have keys for `fromAttribute` and
    /// `toAttribute`. If no `toAttribute` function is provided and
    /// `reflect` is set to `true`, the property value is set directly to the
    /// attribute. A default `converter` is used if none is provided; it supports
    /// `Boolean`, `String`, `Number`, `Object`, and `Array`. Note,
    /// when a property changes and the converter is used to update the attribute,
    /// the property is never updated again as a result of the attribute changing,
    /// and vice versa.
    abstract member converter : (string option -> obj option -> obj) option

    /// Indicates if the property should reflect to an attribute.
    /// If `true`, when the property is set, the attribute is set using the
    /// attribute name determined according to the rules for the `attribute`
    /// property option and the value of the property converted using the rules
    /// from the `converter` property option.
    abstract member reflect : bool option

    /// A function that indicates if a property should be considered changed when
    /// it is set. The function should take the `newValue` and `oldValue` and
    /// return `true` if an update should be requested.
    abstract member hasChanged : (obj -> obj -> bool) option

    /// Indicates whether an accessor will be created for this property. By
    /// default, an accessor will be generated for this property that requests an
    /// update when set. If this flag is `true`, no accessor will be created, and
    /// it will be the user's responsibility to call
    /// `this.requestUpdate(propertyName, oldValue)` to request an update when
    /// the property changes.
    abstract member noAccessor : bool option

/// Map of properties to PropertyDeclaration options.
/// For each property an accessor is made,
/// and the property is processed according to the PropertyDeclaration options.
type PropertyDeclarations = System.Collections.Generic.Dictionary<string, PropertyDeclaration>
```
I chose Dictionary because I think that's compiled as a plain object but using it is a little bit annoying as well